packages/core/test/eventDescriptors.test.ts

```diff
@@ -60,4 +60,157 @@ describe('event descriptors', () => {
     service.send('NEXT');
     expect(service.state.value).toBe('pass');
   });
+
+  it('should NOT support non-tokenized wildcards', () => {
+    const machine = Machine({
+      initial: 'start',
+      states: {
+        start: {
+          on: {
+            'event*': 'success'
```

Comment:
this seems like a clearly incorrect descriptor (SCXML defines it as invalid, interestingly this works in CCXML though and would match `eventwhatever`), should we maybe add dev-only warnings for this?

Reply:
SCXML doesn't define it as invalid directly; it only says that the event name is alphanumeric. However, it only makes this "restriction" (assumedly) for event names, but not for state names or generated IDs.

This restriction can be problematic because a state might nave an ID like `"fetching-user"` and generated event might look like `"done.state.fetching-user"`, which would be invalid if SCXML disallowed it (because of the `"-"` character.

However, this would be contradictory since it's only restrictive for event names.

IMO I would rather be more liberal - event names can consist of any character, but `"."` is the only special character used for tokenization.

Reply:
There are also restrictions for state IDs - they need to be of ID type (defined in the XML Schema spec). But quite frankly I don't care about it, reading that spec would take me more time than i'd like it too and from what I can quickly assess - you are right in a sense that state IDs can have more than alphanumeric characters (for instance I'm pretty sure that `_` is allowed) and that kinda contradicts that statement about alphanumeric restriction for the event names (like you have mentioned).

I'm OK with being more liberal here for other purposes but I think that at least `*` should be considered illegal in the middle of a string. It doesn't do much good and given it has a special meaning when it is on the end it might only end up being confusing if we allow it at any position but without special meaning.

Reply:
> I'm OK with being more liberal here for other purposes but I think that at least * should be considered illegal in the middle of a string.

Agreed, we can add a warning for that.

Reply:
Warning added 👍